apps/ios/SimpleXChat/ChatTypes.swift

```diff
@@ -4334,52 +4355,97 @@ public enum ChatItemTTL: Identifiable, Comparable, Hashable {
     case day
     case week
     case month
+    case year
     case seconds(_ seconds: Int64)
     case none
 
     public static var values: [ChatItemTTL] { [.none, .month, .week, .day] }
 
     public var id: Self { self }
 
-    public init(_ seconds: Int64?) {
+    public init(_ seconds: Int64) {
         switch seconds {
+        case 0: self = .none
         case 86400: self = .day
         case 7 * 86400: self = .week
         case 30 * 86400: self = .month
-        case let .some(n): self = .seconds(n)
-        case .none: self = .none
+        case 365 * 86400: self = .year
+        default: self = .seconds(seconds)
         }
     }
 
-    public var deleteAfterText: LocalizedStringKey {
+    public var deleteAfterText: String {
         switch self {
-        case .day: return "1 day"
-        case .week: return "1 week"
-        case .month: return "1 month"
-        case let .seconds(seconds): return "\(seconds) second(s)"
-        case .none: return "never"
+        case .day: return NSLocalizedString("1 day", comment: "delete after time")
+        case .week: return NSLocalizedString("1 week", comment: "delete after time")
+        case .month: return NSLocalizedString("1 month", comment: "delete after time")
+        case .year: return NSLocalizedString("1 year", comment: "delete after time")
+        case let .seconds(seconds): return String.localizedStringWithFormat(NSLocalizedString("%@ seconds(s)", comment: "delete after time"), seconds)
+        case .none: return NSLocalizedString("never", comment: "delete after time")
         }
     }
 
-    public var seconds: Int64? {
+    public var seconds: Int64 {
         switch self {
         case .day: return 86400
         case .week: return 7 * 86400
         case .month: return 30 * 86400
+        case .year: return 365 * 86400
         case let .seconds(seconds): return seconds
-        case .none: return nil
+        case .none: return 0
         }
     }
 
     private var comparisonValue: Int64 {
-        self.seconds ?? Int64.max
+        if self.seconds == 0 {
+            return Int64.max
+        } else {
+            return self.seconds
+        }
     }
 
     public static func < (lhs: Self, rhs: Self) -> Bool {
         return lhs.comparisonValue < rhs.comparisonValue
     }
 }
 
+public enum ChatTTL: Identifiable, Hashable {
```

Comment:
This type is structurally equivalent to "ChatItemTTL?", so probably unnecessary.

The UI has to show which global default is set, but there is no need to store it in type, it should be just a view concern.

So at a point where you pass the list to ForEach, you would pass [ChatItemTTL?] and render nil as global default.

Then this type can be removed and all its function be local to view(s).